### ISSUE TITLE
Basic Elements

### ISSUE THREAD

**Issue by KenavR** [DevvitIO/Lanflix#8]
Implement some basic elements people can use through out the application.

.lf-pill
.lf-button
.lf-input
.lf-spinner

If you want, you could add additional elements, you think we may need in the future. Whoever works on this can also add different styles. E.g. “.lf-pill.red”, “.lf-pill.blue”,...

**Comment by hotborsh** [comment]
Are there or will be there any style guides for these elements?

**Comment by KenavR** [comment]
We can talk about color schemes and some other basic stuff, but I so far I have nothing specific in mind. We could also look through https://designschool.canva.com/blog/50-meticulous-style-guides-every-startup-see-launching/ find something we like and go from there.

**Comment by teevik** [comment]
Hey, we could maybe use something like Spotify's styling? Or maybe some material design?

**Comment by KenavR** [comment]
Do you mean that we should mirror the styling of Spotify or Material or that we should just use the libraries? I know Material, but does Spotify have a style guide or a CSS "framework"?

**Comment by teevik** [comment]
Meant mirroring, but that's probably a bad idea.  Could use material design? Not an expert on this stuff :P

**Comment by KenavR** [comment]
Actually mirroring is great, it gives a clear reference and we are still building our own. I also really like the Spotify theme.

**Comment by teevik** [comment]
Yeah not sure what to do, we could just make or own styles from the ground

**Comment by teevik** [comment]
Also, will all components have .lf prefix in the class name?

**Comment by KenavR** [comment]
> Now you may ask why are the component(class)name and the directory name different? It is good practice to namespace our components. In our case we prefix every component class with "lf-" for "lanflix". This allows people to use our component library in other projects without interfering with other libraries or project specific CSS.
-[source](https://github.com/DevvitIO/Lanflix/blob/a797c346c437a71d038708473d1186c7eeb22fb2/docs/COMPONENTS.md)

but I am open to discussing that.

**Comment by teevik** [comment]
Yeah, might be a good idea, but don't think people would use this as component library

**Comment by teevik** [comment]
Also, i would love to work on it, could maybe go with some sort of material design style? Think it could really work for something like this

**Comment by KenavR** [comment]
People may not use it as component library, but what if we want to add something like bootstrap in the future. Calling the class "button" may introduce troubles then. 

Material design is great and if you really want to mimic it quite challenging but you can take a look at Material Design Lite to get an idea. (e.g. input field)

**Comment by teevik** [comment]
Yep alright, we can keep the prefix. 

By the way, is it ok to use css variables?

**Comment by KenavR** [comment]
Guess they are fine - https://caniuse.com/#feat=css-variables

**Comment by teevik** [comment]
Alright, will definitely make something like material design easier

**Comment by teevik** [comment]
Will try and see how well material design will work, otherwise we can do something else

**Comment by teevik** [comment]
Hey, is it okay if i try to make a mockup of how the page could look with material design? I'm not good with design programs so would be better if somebody else did it, but i can try

**Comment by KenavR** [comment]
If it helps you sure, I am not too bad with PS but in this case, I would actually hardcode it using Material design lite or something. I am not sure if there are prepared MD elements for PS, but if not I think I would get faster results just hardcoding it in html.

**Comment by teevik** [comment]
We have any sketch of how it's going to be layed out?

**Comment by KenavR** [comment]
Here are some rough sketches. 
![mockup_main](https://user-images.githubusercontent.com/6655009/32699004-042668e4-c7af-11e7-8c9b-4045460428e6.PNG)

![mockup_series](https://user-images.githubusercontent.com/6655009/32699007-0888401a-c7af-11e7-8452-9591547aa98f.PNG)

**Comment by teevik** [comment]
Alright, will start

**Comment by teevik** [comment]
Ah hey, sorry but won't have time to work on this next to school :( Can somebody else work on this issue?

**Comment by teevik** [comment]
Hey, sorry for being away for so long

Any thoughts on this? https://xd.adobe.com/view/9326c9bc-ed4e-4f94-9d2f-08ed6d7f116a (or zip) [lanflix.zip](https://github.com/DevvitIO/Lanflix/files/1819701/lanflix.zip)

It's pretty much like the sketch, but added some shadows, etc. There's only the home page right now

**Comment by KenavR** [comment]
Looks great. Sorry, for the delay I am scheduling my days now and I will set aside time for this project.
